crepecake/src/main/java/com/google/cloud/tools/crepecake/image/Digest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a SHA-256 digest as defined by the Registry HTTP API v2 reference.
+ *
+ * @see <a
+ *     href="https://docs.docker.com/registry/spec/api/#content-digests">https://docs.docker.com/registry/spec/api/#content-digests</a>
+ */
+public class Digest {
+
+  /** Pattern matches a SHA-256 hash - 32 bytes in lowercase hexadecimal. */
+  private static final String HASH_REGEX = "[a-f0-9]{64}";
+
+  /** Pattern matches a SHA-256 digest - a SHA-256 hash prefixed with "sha256:". */
+  private static final String DIGEST_REGEX = "sha256:" + HASH_REGEX;
+
+  private static final Pattern HASH_PATTERN = Pattern.compile(HASH_REGEX);
+
+  private final String digest;
```

Comment:
If I read the spec right, digest is always "sha256:" + hash. Given that, it seems iffy to store both. Might want to store only the hash and provide a simple getter method for the digest form